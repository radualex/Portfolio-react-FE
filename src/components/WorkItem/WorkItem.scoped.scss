#container {
  height: 30%;
  #wrapper {
    display: flex;
    height: 100%;
    img {
      // width: 30rem;
      // height: 14rem;
      // height: auto;
      // width: auto;
      margin: 0 1rem 0.8rem 0;
      border-radius: 1rem;
      object-fit: contain;
      max-width: 100%;
      max-height: 100%;
    }
  }

  .separator {
    border: 0.1px solid #e0e0e0;
    margin: 0 0 1rem 0;
  }
}
