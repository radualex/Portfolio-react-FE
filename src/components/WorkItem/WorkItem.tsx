import React from "react";
import me from "../../assets/me.png";

import "./WorkItem.scoped.scss";

export const WorkItem = () => {
  return (
    <div id="container">
      <div id="wrapper">
        <img src={me} alt="project" />
        <div className="content">
          <h2 className="content__title">Hello!</h2>
          <div className="content__header">
            <span className="content__header__year">2021</span>
            <span className="content__header__project">Dashboard</span>
          </div>
          <p className="content__description">
            Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet
            sint. Velit officia consequat duis enim velit mollit. Exercitation
            veniam consequat sunt nostrud amet.
          </p>
        </div>
      </div>
      <hr className="separator" />
    </div>
  );
};
