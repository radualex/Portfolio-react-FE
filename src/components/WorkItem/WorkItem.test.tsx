import React from 'react';
import { render, screen } from "@testing-library/react";

import { WorkItem } from "./WorkItem";

describe("WorkItem", () => {
  it("renders title", () => {
    render(<WorkItem />);
    const title = screen.getByText("Hello!");
    expect(title).toBeInTheDocument();
  });

  it("renders work item", () => {
    const { asFragment } = render(<WorkItem />);
    expect(asFragment()).toMatchSnapshot();
  });
});
