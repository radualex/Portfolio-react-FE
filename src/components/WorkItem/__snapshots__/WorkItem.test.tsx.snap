// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkItem renders work item 1`] = `
<DocumentFragment>
  <div
    id="container"
  >
    <div
      id="wrapper"
    >
      <img
        alt="project"
        src="me.png"
      />
      <div
        class="content"
      >
        <h2
          class="content__title"
        >
          Hello!
        </h2>
        <div
          class="content__header"
        >
          <span
            class="content__header__year"
          >
            2021
          </span>
          <span
            class="content__header__project"
          >
            Dashboard
          </span>
        </div>
        <p
          class="content__description"
        >
          Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit. Exercitation veniam consequat sunt nostrud amet.
        </p>
      </div>
    </div>
    <hr
      class="separator"
    />
  </div>
</DocumentFragment>
`;
